pkg/controllermanager/controller/event/event_control_test.go

```diff
@@ -0,0 +1,201 @@
+// limitations under the License.
+
+package event
+
+import (
+	"context"
+	"time"
+
+	fakeclientmap "github.com/gardener/gardener/pkg/client/kubernetes/clientmap/fake"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/logger"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("#reconcileEvent", func() {
+	var (
+		ctrl                   *gomock.Controller
+		clientMap              *fakeclientmap.ClientMap
+		k8sGardenRuntimeClient *mockclient.MockClient
+
+		namespaceName = "garden-foo"
+		requestName   = "request"
+
+		ttl = &metav1.Duration{Duration: 1 * time.Hour}
+
+		eventController            *Controller
+		request                    reconcile.Request
+		shootEvent                 *corev1.Event
+		nonShootEvent              *corev1.Event
+		eventWithoutInvolvedObject *corev1.Event
+		cfg                        *config.ControllerManagerConfiguration
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		k8sGardenRuntimeClient = mockclient.NewMockClient(ctrl)
+		clientMap = fakeclientmap.NewClientMap().AddRuntimeClient(keys.ForGarden(), k8sGardenRuntimeClient)
+
+		logger.Logger = logger.NewNopLogger()
+
+		shootEvent = &corev1.Event{
+			FirstTimestamp: metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Shoot"},
+		}
+		nonShootEvent = &corev1.Event{
+			FirstTimestamp: metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Project"},
+		}
+		eventWithoutInvolvedObject = &corev1.Event{
+			FirstTimestamp: metav1.Time{Time: time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+		}
+		request = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: namespaceName,
+				Name:      requestName,
+			}}
+		cfg = &config.ControllerManagerConfiguration{
+			Controllers: config.ControllerManagerControllerConfiguration{
+				Event: &config.EventControllerConfiguration{
+					TTLNonShootEvents: ttl,
+				},
+			},
+		}
+
+		eventController = NewController(clientMap, cfg)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Context("Shoot Events", func() {
+
+		It("should ignore them", func() {
+			mockClientGet(k8sGardenRuntimeClient, request.NamespacedName, shootEvent)
+			k8sGardenRuntimeClient.EXPECT().Delete(gomock.Any(), gomock.Any()).Times(0)
+			expected := reconcile.Result{}
+			actual, err := eventController.reconcileEvent(request)
+			Expect(err).To(BeNil())
```

Comment:
Please use `HaveOccurred` for matching errors. It also checks, that `actual` is really an `error` and not just `nil`. This avoids making incorrect assertions :)
```suggestion
			Expect(err).NotTo(HaveOccurred())
```